### PR TITLE
Add Product tests for OAuth2 with Proxy

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

**Review by sergey-melnychuk — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java [hunk 1]
```diff
@@ -32,14 +36,15 @@
...
     private final DockerFiles.ResourceProvider configDir;
 
     @Inject
     public HydraIdentityProvider(PortBinder binder, DockerFiles dockerFiles)
     {
         this.binder = requireNonNull(binder, "binder is null");
-        requireNonNull(dockerFiles, "dockerFiles is null");
-        this.configDir = dockerFiles.getDockerFilesHostDirectory("common/hydra-identity-provider");
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        this.configDir = this.dockerFiles.getDockerFilesHostDirectory("common/hydra-identity-provider");
```
Comment (lukasz-walkiewicz) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java:47]: nit: I would say `this` is redundant here.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java [hunk 2]
```diff
@@ -88,6 +93,19 @@ public void extendEnvironment(Environment.Builder builder)
...
+            if (isPrestoContainer(dockerContainer.getLogicalName())) {
+                dockerContainer
+                        .withCopyFileToContainer(
+                                forHostPath(configDir.getPath("cert")),
+                                CONTAINER_PRESTO_ETC + "/hydra/cert");
+            }
+        });
+
+        builder.configureContainer(TESTS, dockerContainer -> dockerContainer.withCopyFileToContainer(
+                forHostPath(dockerFiles.getDockerFilesHostPath("conf/tempto/tempto-configuration-for-docker-oauth2.yaml")),
```
Comment (lukasz-walkiewicz) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java:107]: Wouldn't it be easier to just more `tempto-configuration-for-docker-oauth2.yaml` to `common/hydra-identity-provider`?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        DockerContainer proxy = new DockerContainer("httpd:2.4.51", "proxy");
+        proxy.withCopyFileToContainer(forHostPath(configDir.getPath("httpd.conf")), "/usr/local/apache2/conf/httpd.conf");
+        builder.addContainer(proxy);
+        builder.containerDependsOn(COORDINATOR, "proxy");
+    }
+
+    @Override
+    public List<EnvironmentExtender> getDependencies()
+    {
+        return ImmutableList.of(hydraIdentityProvider);
```
Comment (lukasz-walkiewicz) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvSinglenodeOauth2HttpProxy.java:88]: Is this required when `hydraIdentityProvider` is already in `super(ImmutableList.of(standard, hydraIdentityProvider, seleniumChrome));`?

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy/cert/create-certs.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -eux
+
+for i in hydra proxy; do
+    echo $i
+
+    # Generate private key and certificate
+    openssl req -new -x509 -newkey rsa:4096 -sha256 -nodes -keyout $i.key -days 3560 -out $i.crt -config $i.conf
```
Comment (lukasz-walkiewicz) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2-https-proxy/cert/create-certs.sh:9]: 3650? ;)

**Review by lukasz-walkiewicz — COMMENTED** [review]
looks good, minor comments

**Comment by Praveen2112** [comment]
@lukasz-walkiewicz AC

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteOauth2.java [hunk 5]
```diff
@@ -35,6 +36,9 @@
                         .withGroups("oauth2")
                         .build(),
                 testOnEnvironment(EnvSinglenodeOauth2HttpProxy.class)
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteOauth2.java:38]: so maybe now we can remove non secured proxy?

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Thank you!

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteOauth2.java (hunk 5 above)
Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/SuiteOauth2.java:38]: I guess it better to run the tests on both the Environment (with non-secure and secure proxy).

**Comment by findepi** [comment]
see https://github.com/trinodb/trino/runs/4336555420?check_suite_focus=true (https://github.com/trinodb/trino/pull/10089)

```
presto-master       | 1) [Guice/ErrorInjectingConstructor]: RuntimeException: Error reading JWK keys from https://hydra:4444/.well-known/jwks.json
presto-master       |   at JwkService.<init>(JwkService.java:61)
presto-master       |   at GeneratedMethodAccessor26.invoke(Unknown Source)
presto-master       |       \_ installed by: ServerSecurityModule -> ConditionalModule -> ConfigurationAwareModule$1 -> OAuth2AuthenticationSupportModule -> OAuth2ServiceModule
presto-master       |   at JwkSigningKeyResolver.<init>(JwkSigningKeyResolver.java:34)
presto-master       |       \_ for 1st parameter keys
presto-master       |   while locating JwkSigningKeyResolver
presto-master       |   at GeneratedMethodAccessor26.invoke(Unknown Source)
presto-master       |       \_ installed by: ServerSecurityModule -> ConditionalModule -> ConfigurationAwareModule$1 -> OAuth2AuthenticationSupportModule -> OAuth2ServiceModule
presto-master       |   at OAuth2Service.<init>(OAuth2Service.java:108)
presto-master       |       \_ for 2nd parameter signingKeyResolver
presto-master       |   at GeneratedMethodAccessor26.invoke(Unknown Source)
presto-master       |       \_ installed by: ServerSecurityModule -> ConditionalModule -> ConfigurationAwareModule$1 -> OAuth2AuthenticationSupportModule -> OAuth2ServiceModule
presto-master       |   at OAuth2Authenticator.<init>(OAuth2Authenticator.java:43)
presto-master       |       \_ for 1st parameter service
presto-master       |   while locating OAuth2Authenticator
presto-master       |   at ServerSecurityModule.lambda$authenticatorModule$5(ServerSecurityModule.java:109)
presto-master       |       \_ installed by: ServerSecurityModule -> ConditionalModule -> ConfigurationAwareModule$1 -> ServerSecurityModule$$Lambda$475/0x00000001002d5c40
presto-master       |   while locating Authenticator annotated with @Element(setName=,uniqueId=429, type=MAPBINDER, keyType=String)
presto-master       |   at ServerSecurityModule.getAuthenticatorList(ServerSecurityModule.java:95)
presto-master       |       \_ for 2nd parameter authenticators
presto-master       |   at ServerSecurityModule.getAuthenticatorList(ServerSecurityModule.java:95)
presto-master       |   at AuthenticationFilter.<init>(AuthenticationFilter.java:54)
presto-master       |       \_ for 1st parameter authenticators
presto-master       |   at ResourceSecurityDynamicFeature.<init>(ResourceSecurityDynamicFeature.java:70)
presto-master       |       \_ for 2nd parameter authenticationFilter
presto-master       |   at ServerSecurityModule.setup(ServerSecurityModule.java:56)
presto-master       |   while locating ResourceSecurityDynamicFeature
presto-master       |   at JaxrsBinder.bind(JaxrsBinder.java:37)
presto-master       |   while locating Object annotated with @Element(setName=@JaxrsResource,uniqueId=415, type=MULTIBINDER, keyType=)
presto-master       |   at JaxrsModule$JaxRsApplication.<init>(JaxrsModule.java:92)
presto-master       |       \_ for 1st parameter jaxRsSingletons
presto-master       |   while locating JaxrsModule$JaxRsApplication
presto-master       | 	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
presto-master       | 	at com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.get(InternalProviderInstanceBindingImpl.java:164)
presto-master       | 	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:60)
presto-master       | 	at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:40)
presto-master       | 	at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:60)
presto-master       | 	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
presto-master       | 	at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:33)
presto-master       | 	at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:98)
presto-master       | 	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:109)
presto-master       | 	at io.airlift.bootstrap.LifeCycleModule.provision(LifeCycleModule.java:54)
presto-master       | 	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:117)
presto-master       | 	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
presto-master       | 	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
presto-master       | 	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:296)
presto-master       | 	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:58)
presto-master       | 	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
presto-master       | 	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:169)
presto-master       | 	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45)
presto-master       | 	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
presto-master       | 	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:186)
presto-master       | 	... 8 more
presto-master       | Caused by: java.lang.NullPointerException
presto-master       | 	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.onUpgradeFrom(HttpReceiverOverHTTP.java:123)
presto-master       | 	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onUpgradeFrom(HttpConnectionOverHTTP.java:179)
presto-master       | 	at org.eclipse.jetty.io.AbstractEndPoint.upgrade(AbstractEndPoint.java:432)
presto-master       | 	at org.eclipse.jetty.client.HttpProxy$CreateTunnelPromise.tunnelSucceeded(HttpProxy.java:220)
presto-master       | 	at org.eclipse.jetty.client.HttpProxy$CreateTunnelPromise.lambda$tunnel$0(HttpProxy.java:189)
presto-master       | 	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:218)
presto-master       | 	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:210)
presto-master       | 	at org.eclipse.jetty.client.HttpSender.terminateRequest(HttpSender.java:411)
presto-master       | 	at org.eclipse.jetty.client.HttpSender.someToSuccess(HttpSender.java:331)
presto-master       | 	at org.eclipse.jetty.client.HttpSender$CommitCallback.process(HttpSender.java:744)
presto-master       | 	at org.eclipse.jetty.client.HttpSender$CommitCallback.succeeded(HttpSender.java:710)
presto-master       | 	at org.eclipse.jetty.client.http.HttpSenderOverHTTP$HeadersCallback.onCompleteSuccess(HttpSenderOverHTTP.java:307)
presto-master       | 	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:326)
presto-master       | 	at org.eclipse.jetty.util.IteratingCallback.succeeded(IteratingCallback.java:366)
presto-master       | 	at org.eclipse.jetty.client.http.HttpSenderOverHTTP$HeadersCallback.succeeded(HttpSenderOverHTTP.java:293)
presto-master       | 	at org.eclipse.jetty.io.WriteFlusher.completeWrite(WriteFlusher.java:394)
presto-master       | 	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.lambda$fill$1(SslConnection.java:835)
presto-master       | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
presto-master       | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
presto-master       | 	at java.base/java.lang.Thread.run(Thread.java:829)
presto-master       | 
presto-master       | 
presto-master       | (exited)
```


can this be related to this PR?

cc @losipiuk

**Pull request merged**